### PR TITLE
Make serviceManager.isolateManager source of truth for current isolate.

### PR DESCRIPTION
**Pull request by polina-c**
RELEASE_NOTE_EXCEPTION=[refactoring]

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart [hunk 1]
```diff
@@ -170,9 +170,7 @@ class DebuggerController extends DisposableController
     // and modify to detect if app is paused from the isolate
     // https://github.com/flutter/devtools/pull/4993#discussion_r1060845351
 
-    serviceManager.appState
-      ..setIsolateRef(ref)
-      ..setPausedOnBreakpoint(false);
+    serviceManager.appState..setPausedOnBreakpoint(false);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart:173]: remove the .. now that there is only one method call

File: packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart [hunk 2]
```diff
@@ -540,7 +538,7 @@ class DebuggerController extends DisposableController
         .map(
           (v) => DartObjectNode.create(
             v,
-            serviceManager.appState.isolateRef.value,
+            serviceManager.isolateManager.selectedIsolate.value,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart:541]: optional - we could add a convenience getter to service manager:
`IsolateRef? get selectedIsolate => isolateManager.selectedIsolate.value;` and use that everywhere for brevity

File: packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart (hunk 1 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart]: Filed https://github.com/dart-lang/linter/issues/3960 to consider adding a lint to handle to warn about this case.

**Review by jacob314 — APPROVED** [review]
lgtm

**Pull request merged**